obs-studio, libcef: support aarch64

lf- opened the pull request:
~~All functionality is supported except for the OBS browser, which does not work due to our libcef being broken on aarch64. I don't think this is inherent, Chromium definitely works on aarch64, it's probably a packaging/other thing. Regardless that's outside the scope of my work here.~~

~~I also marked libcef x86_64-linux only because it seems to be doing x86 specific stuff.
This is probably a bug in that package.~~

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->


###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`) thanks @Hoverbear for testing
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

lf- committed:
obs-studio: add aarch64 support

lf- committed:
libcef: add aarch64 support

lf- commented:
cc @puffnfresh as maintainer of libcef: just so you're aware, libcef only builds on x86_64. It can probably be fixed; it appears spotify offers libcef builds for other architectures and the nix expression is just hardcoded to grab the x86_64 one.

Hoverbear commented:
I was able to build and run this successfully on my aarch64 Honeycomb LX2K! Great job. :)

r-rmcgibbo commented:
Result of `nixpkgs-review pr 133428` at d158a01e run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/87c804b74c24918ac3826590004225bb">obs-studio-plugins.obs-ndi</a></li>
  </ul>
</details>
<details>
  <summary>8 packages built successfully:</summary>
  <ul>
    <li>obs-studio</li>
    <li>obs-studio-plugins.looking-glass-obs</li>
    <li>obs-studio-plugins.obs-gstreamer</li>
    <li>obs-studio-plugins.obs-move-transition</li>
    <li>obs-studio-plugins.obs-multi-rtmp</li>
    <li>obs-studio-plugins.obs-nvfbc</li>
    <li>obs-studio-plugins.obs-websocket</li>
    <li>obs-studio-plugins.wlrobs</li>
  </ul>
</details>
<details>
  <summary>3 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

makeWrapper is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/video/obs-studio/default.nix:74:3:
```
   |
74 |   buildInputs = [
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl2` is a deprecated license, please check if project uses `gpl2Plus` or `gpl2Only` and change `meta.license` accordingly.

Near pkgs/applications/video/obs-studio/default.nix:143:5:
```
    |
143 |     license = licenses.gpl2;
    |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/build-tools-in-build-inputs.md">build-tools-in-build-inputs</a>

swig is a build tool so it likely goes to `nativeBuildInputs`, not `buildInputs`.

Near pkgs/applications/video/obs-studio/default.nix:74:3:
```
   |
74 |   buildInputs = [
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

lf- commented:
> 1 package failed to build:

Context: This is a package that is broken on purpose in order to comply with a nonfree license and is not anything to do with my work.

----

I have redone the patch to also fix libcef so aarch64 can have a browser with its OBS, because it was not terribly hard and I was feeling like shaving yaks. @Hoverbear can you have another go at testing it?

lf- committed:
libcef: support i686

Previously, libcef did not actually support i686!! It compiled against a
64 bit chromium, which may have actually worked on CI but it would not
work so well on an actual 32 bit machine.

SuperSandro2000 commented on pkgs/applications/video/obs-studio/default.nix:
```diff
@@ -61,7 +64,13 @@ in mkDerivation rec {
+    cmake
+    pkg-config
+    makeWrapper
+  ]
+  ++ optional scriptingSupport swig;
```
```suggestion
  ] ++ optional scriptingSupport swig;
```
